src/client/src/StreamState.c

```diff
@@ -705,14 +705,34 @@ STATUS executeStoppedStreamState(UINT64 customData, UINT64 time)
     ENTERS();
     UNUSED_PARAM(time);
     STATUS retStatus = STATUS_SUCCESS;
+    UINT64 duration, viewByteSize;
     PKinesisVideoStream pKinesisVideoStream = STREAM_FROM_CUSTOM_DATA(customData);
 
     CHK(pKinesisVideoStream != NULL, STATUS_NULL_ARG);
 
     // Also store the connection stopping result
     pKinesisVideoStream->connectionDroppedResult = pKinesisVideoStream->base.result;
 
-    // Step the client state machine first
+    // Check if we want to prime the state machine based on whether we have any more content to send
+    // currently and if the error is a timeout.
+    if (SERVICE_CALL_RESULT_IS_A_TIMEOUT(pKinesisVideoStream->connectionDroppedResult)) {
```

Comment:
Did you consider putting this logic inside streamTerminatedEvent? I think any kind of timeout would have to go through that path. In there we can simply do if viewByteSize == 0 then resetStream.

Reply:
streamTerminatedEvent is only applicable for the upload handles - i,e, when we are streaming already. It actually does all the right things. The differentiation is to not auto-prime the state machine which is what we are doing here.

Reply:
Also, resetStream is a very tough measure with dumping the frame buffers. It's also kind of orthogonal - the main idea here is to do "idling"